### PR TITLE
redis replication check (for master or slave)

### PR DESCRIPTION
**Pull request by ghost**
example:

<pre>
$ ./redis_replication_check.rb --help
Checks redis replication status.
    -h, --host <host>                connect to redis running on <host>.  defaults to localhost
    -p, --port <port>                connect to redis running on <port>.  defaults to 6379
    -r, --role <master|slave>        redis should be master or slave
    -c, --connected_slaves <int>     master should have <int> slaves.  must be used with -r/--role master.  defaults to 1
    -l, --replication_lag <int>      master io cannot be more than <int> seconds behind.  used with -r/--role slave.  default is 0
        --help                       Display this screen

$ ./redis_replication_check.rb -r slave
status ok ok
metric master_last_io_seconds_ago int 0

$ ./redis_replication_check.rb -r slave
status err redis thinks it is slave, master is up and last io was 3s.
metric master_last_io_seconds_ago int 3

$ ./redis_replication_check.rb -r slave -l 5
status ok ok
metric master_last_io_seconds_ago int 3

$ ./redis_replication_check.rb -h someremotehost -r master -c 10
status err redis thinks it is master with 1 slaves.

$ ./redis_replication_check.rb -h someremotehost -r master -c 1
status ok ok

</code>

### CODE REVIEW

**Comment by flores** [comment]
(added change to some other scripts that said "critical" instead of "err" on status because github did it for me :).  let me know if you prefer a second pull request!)

File: redis_replication_check.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+		puts "status ok ok"
+	else
+		puts "status err redis thinks it is #{status["role"]}, master is #{status["master_link_status"]} and last io was #{status["master_last_io_seconds_ago"]}s."
+	end
+	puts "metric master_last_io_seconds_ago int #{status["master_last_io_seconds_ago"]}"	
+else
+	expectedslaves = options[:connected_slaves] || 1
+	actualslaves = status["connected_slaves"].to_i
+	if status["role"] =~ /master/ && actualslaves == expectedslaves
+
```
Comment (Kami) [inline redis_replication_check.rb:103]: Here you could also print the `role` and `connected_slaves` metric.

Otherwise the pull requests looks fine.

**Comment by flores** [comment]
Commited Kami.  Thanks!

**Comment by Kami** [comment]
Merged, thanks.

**Pull request merged**